codegen/utils.py

```diff
@@ -30,6 +31,17 @@ def bool_to_c_str(b: bool) -> str:
   return str(b).lower()
 
 
+def split_into_chunks(
+    data: Iterable[Any],
+    chunk_size: int) -> Generator[Tuple[Any, ...], None, None]:
+  """Splits an iterable into chunks of a given size."""
+  data_iterator = iter(data)
+  chunk = tuple(itertools.islice(data_iterator, chunk_size))
+  while chunk:
+    yield chunk
+    chunk = tuple(itertools.islice(data_iterator, chunk_size))
```

Comment:
this reads a bit funny.

Is there any way to write this where the loop looks something like:
```py
while <not sure what the condition would be>:
    chunk = tuple(itertools.islice(data_iterator, chunk_size))
    yield chunk
```

Reply:
WDYT of this:

```
while True:
  chunk = tuple(itertools.islice(data_iterator, chunk_size))
  if not chunk:
    break
  yield chunk
```

Reply:
+1. that seems  more readable to me.

Reply:
Done